tests/scripts/generate_psa_tests.py

```diff
@@ -78,6 +78,11 @@ def automatic_dependencies(*expressions: str) -> List[str]:
     """
     used = set()
     for expr in expressions:
+        # HKDF_EXTRACT and HKDF_EXPAND algs depend on HKDF
```

Comment:
I can see that `PSA_WANT_ALG_HKDF` is used as dependency in test and  `#if` guards use `MBEDTLS_PSA_BUILTIN_ALG_HKDF`. Do I need to add also `MBEDTLS_PSA_BUILTIN_ALG_HKDF_EXTRACT/EXPAND` and add guards for related code?

Reply:
Yes please. We should treat all algorithms the same way unless there's a very good reason: for us it simplifies reasoning about the code and testing, and for users it simplifies the understanding of how to use the library (which also reduces our support load).

Reply:
I understand that both should be still defined under `MBEDTLS_HKDF_C`:

```
#if defined(MBEDTLS_HKDF_C)
#define MBEDTLS_PSA_BUILTIN_ALG_HMAC 1
#define PSA_WANT_ALG_HMAC 1
#define MBEDTLS_PSA_BUILTIN_ALG_HKDF 1
+#define MBEDTLS_PSA_BUILTIN_ALG_HKDF_EXTRACT 1
+#define MBEDTLS_PSA_BUILTIN_ALG_HKDF_EXPAND 1
#define PSA_WANT_ALG_HKDF 1
+#define PSA_WANT_ALG_HKDF_EXTRACT 1
+#define PSA_WANT_ALG_HKDF_EXPAND 1
#endif /* MBEDTLS_HKDF_C */
```

or we need also `MBEDTLS_HKDF_EXTRACT_C` and `MBEDTLS_HKDF_EXPAND_C`?

Reply:
Actually the PSA HKDF implementation does not use `hkdf.c`. So that part of `include/mbedtls/config_psa.h` should not have been there at all! The PSA HKDF can be available whenever the underlying hash is. https://github.com/Mbed-TLS/mbedtls/issues/5890 (please don't fix it here, but don't add HKDF_EXxxx there either)